src/app/screens/connect/btcSelectAddressScreen/index.tsx

```diff
@@ -0,0 +1,292 @@
+  width: 328,
+  height: 66,
+  padding: props.theme.spacing(10),
+  display: 'flex',
+  flexDirection: 'row',
+  alignItems: 'center',
+  justifyContent: 'space-between',
+  backgroundColor: props.theme.colors.elevation6_800,
+  marginBottom: 1,
+  ':first-of-type': {
+    borderTopLeftRadius: props.theme.radius(2),
+    borderTopRightRadius: props.theme.radius(2),
+  },
+  ':last-of-type': {
+    borderBottomLeftRadius: props.theme.radius(2),
+    borderBottomRightRadius: props.theme.radius(2),
+  },
+}));
+
+const AddressContainer = styled.div({
+  display: 'flex',
+  alignItems: 'center',
+});
+
+const TopImage = styled.img({
+  maxHeight: 32,
+  maxWidth: 32,
+});
+
+const Title = styled.h1((props) => ({
+  ...props.theme.typography.body_bold_l,
+  color: props.theme.colors.white_0,
+  marginTop: 12,
+}));
+
+const DapURL = styled.h2((props) => ({
+  ...props.theme.typography.body_medium_m,
+  color: props.theme.colors.white_400,
+  marginTop: props.theme.spacing(2),
+  textAlign: 'center',
+}));
+
+const AddressImage = styled.img({
+  width: 24,
+  height: 24,
+  marginRight: 8,
+});
+
+const AddressTextTitle = styled.h2((props) => ({
+  ...props.theme.typography.body_medium_m,
+  color: props.theme.colors.white_200,
+  textAlign: 'center',
+}));
+
+const TruncatedAddress = styled.h3((props) => ({
+  ...props.theme.typography.body_medium_m,
+  color: props.theme.colors.white_0,
+  textAlign: 'right',
+}));
+
+const BnsName = styled.h3((props) => ({
+  ...props.theme.typography.body_medium_m,
+  color: props.theme.colors.white_0,
+  textAlign: 'right',
+}));
+
+const RequestMessage = styled.p((props) => ({
+  ...props.theme.typography.body_medium_m,
+  color: props.theme.colors.white_200,
+  textAlign: 'left',
+  wordWrap: 'break-word',
+  marginTop: props.theme.spacing(12),
+  marginBottom: props.theme.spacing(12),
+}));
+
+const PermissionsTitle = styled.h3((props) => ({
+  ...props.theme.typography.body_medium_m,
+  color: props.theme.colors.white_200,
+  textAlign: 'left',
+  marginTop: 24,
+}));
+
+const Permission = styled.div((props) => ({
+  ...props.theme.typography.body_medium_m,
+  color: props.theme.colors.white_0,
+  marginTop: 12,
+  display: 'flex',
+  alignContent: 'center',
+}));
+
+const PermissionIcon = styled.div({
+  marginRight: 4,
+});
+
+const ActionsContainer = styled(StickyHorizontalSplitButtonContainer)({
+  marginTop: 'auto',
+});
+
+function BtcSelectAddressScreen() {
+  const [loading, setLoading] = useState(false);
+  const navigate = useNavigate();
+  const theme = useTheme();
+  const { t } = useTranslation('translation', { keyPrefix: 'SELECT_BTC_ADDRESS_SCREEN' });
+  const { network, btcAddress, ordinalsAddress, stxAddress, selectedAccount } = useWalletSelector();
+  const [appIcon, setAppIcon] = useState<string>('');
+  const { payload, origin, approveBtcAddressRequest, cancelAddressRequest } =
+    useBtcAddressRequest();
+  const appUrl = useMemo(() => origin.replace(/(^\w+:|^)\/\//, ''), [origin]);
```

Comment:
useMemo feels like overkill here for a regex? you instantiate an extra function and deps array plus add the entire useMemo code path